app/controllers/orders/client_summary.js

```diff
@@ -19,7 +23,60 @@ export default detail.extend({
     return this.get("store").peekAll("identity_type");
   }),
 
+  isLongDescription: Ember.computed("model.purposeDescription", function() {
+    return this.get("model.purposeDescription.length") > LONG_DESCRIPTION_SIZE;
+  }),
+
+  toggleDisplayShowMore: Ember.observer("model.purposeDescription", function() {
+    const isLongDescription = this.get("isLongDescription");
+    isLongDescription
+      ? this.set("displayShowMore", true)
+      : this.set("displayShowMore", false);
+  }),
+
+  shortPurposeDescription: Ember.computed(
+    "showInlineEdit",
+    "displayShowMore",
+    function() {
+      let purposeDescription = this.get("model.purposeDescription");
+      if (this.get("displayShowMore")) {
+        return purposeDescription.substring(0, LONG_DESCRIPTION_SIZE) + "...";
+      } else {
+        return purposeDescription;
+      }
+    }
+  ),
+
   actions: {
+    showMoreToggle() {
+      this.toggleProperty("displayShowMore");
+    },
+
+    enableDescriptionEdit(opts = {}) {
+      const { makeEditable = true } = opts;
+      this.set("showInlineEdit", makeEditable);
+      if (makeEditable) {
+        Ember.run.later(() => {
+          Ember.$("#desc").click();
+        }, 50);
+      }
+    },
+
+    updatePurposeDescription(purposeDescription) {
+      const order = this.get("model");
+      this.showLoadingSpinner();
+      order.set("purposeDescription", purposeDescription);
+      order
+        .save()
+        .then(() => {
+          Ember.$(".client_summary_description_error").hide();
```

Comment:
we want to avoid using jquery as a general rule of thumb, and rather use a property with and `#if` hbs condition

`set('showError', true)`

Same goes for setting/unsetting classes, examples available [here](https://guides.emberjs.com/release/templates/conditionals/)